packages/ruby/lib/readme/filter.rb

```diff
@@ -11,13 +11,26 @@ def self.for(reject: nil, allow_only: nil)
     end
   end
 
+  def self.redact(rejected_params)
+    rejected_params.each_with_object({}) do |(k, v), hash|
+      # If it's a string then return the length of the redacted field
+      hash[k.to_str] = "[REDACTED#{v.instance_of?(String) ? " #{v.length}" : ""}]"
+    end
+  end
+
   class AllowOnly
-    def initialize(filter_values)
-      @filter_values = filter_values.map(&:downcase)
+    def initialize(filter_fields)
+      @allowed_fields = filter_fields
     end
 
     def filter(hash)
-      hash.select { |key, _value| @filter_values.include?(key.downcase) }
+      allowed_fields = @allowed_fields.map(&:downcase)
+      # Select all fields that are allowed
+      allowed_params = hash.select { |key, _value| allowed_fields.include?(key.downcase) }
+      # If a field is not in the allowed fields reject it
+      rejected_params = hash.reject { |key, _value| allowed_fields.include?(key.downcase) }
```

Comment:
[`Enumerable#partition`](https://ruby-doc.org/core-2.5.1/Enumerable.html#method-i-partition) could do this in one loop?

Reply:
or maybe [Enumerable#each_with_object](https://ruby-doc.org/core-2.1.1/Enumerable.html#method-i-each_with_object)